kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/BigQuerySinkTask.java

```diff
@@ -303,10 +300,13 @@ private boolean recordEmpty(SinkRecord record) {
 
   private Partitioner<RowToInsert> getPartitioner() {
     int maxWriteSize = config.getInt(config.MAX_WRITE_CONFIG);
+    BigQueryWriter writer = getWriter();
     if (maxWriteSize == -1) {
-      return new SinglePartitioner<>();
+      return new SinglePartitioner(writer);
+    } else if (maxWriteSize == 0) {
+      return new DynamicPartitioner(writer);
     } else {
-      return new EqualPartitioner<>(maxWriteSize);
+      return new EqualPartitioner(writer, maxWriteSize);
```

Comment:
I'm in favor of just deleting equal partitioner and having either single or dynamic. Also in favor of eliminating maxWriteSize, and just having a config to define which partitioner to use single, or dynamic.

Thoughts?